### PR TITLE
Support StateProof keys.

### PR DESCRIPTION
**Pull request by shiqizng**
## Summary

Add new field, StateProofKey, to key reg txn fields and account participation. Accounts data written to postgres is also updated accordingly to include the additional field. 

## Test Plan
Add new unit and e2e tests for api handler to validate that StateProofKey is returned correctly in response.

### CODE REVIEW

**Review by jasonpaulos — COMMENTED** [review]
Generally looks good to me, though I can't give great feedback on some indexer-specific testing changes, like if it's ok to modify `createTxn`.

Also note that the go-algorand submodule was updated to use the branch from https://github.com/algorand/go-algorand/pull/2990, so probably that needs to be merged and the submodule changed back to master (or stable?) before this PR should be merged

File: api/handlers_e2e_test.go
Comment (tolikzinovyev) [inline api/handlers_e2e_test.go:428]: please delete empty lines

File: api/handlers_e2e_test.go
Comment (tolikzinovyev) [inline api/handlers_e2e_test.go:359]: please delete

File: api/handlers_e2e_test.go
Comment (tolikzinovyev) [inline api/handlers_e2e_test.go:373]: This is interesting. Is there a reason to compare json encodings instead of byte arrays?

File: api/handlers_e2e_test.go
Comment (tolikzinovyev) [inline api/handlers_e2e_test.go:353]: I would put this and next section in its own scope so that the section after doesn't accidentally use any of the old variables.
```
{
  req := httptest.NewRequest(http.MethodGet, "/", nil)
  ...
  require.Equal(t, json.Encode(sprfkey), json.Encode(*response.Transaction.KeyregTransaction.StateProofKey))
}
```

File: api/handlers_e2e_test.go
Comment (tolikzinovyev) [inline api/handlers_e2e_test.go:332]: It would be nice to rename `sprfkey` to `stateProofPK` for consistency with `votePK` and `selectionPK`.

**Review by tolikzinovyev — CHANGES_REQUESTED** [review]
Thanks for the PR. The code looks good overall, but the compact cert transaction type needs to be added in `idb/txn_type_enum.go` as discussed.

**Comment by shiqizng** [comment]
> Thanks for the PR. The code looks good overall, but the compact cert transaction type needs to be added in `idb/txn_type_enum.go` as discussed.

compact cert will be part of a different release. I was told to include keyreg changes only.

File: api/handlers_e2e_test.go
Comment (shiqizng) [reply api/handlers_e2e_test.go]: updated.

File: api/handlers_e2e_test.go
Comment (shiqizng) [reply api/handlers_e2e_test.go]: sprfKey type merklekeystore.Verifier and response.Transaction.KeyregTransaction.StateProofKey is deserialized to *byte[].  This check fails if I don't convert them to the same type.

Comment (jasonpaulos) [reply api/handlers_e2e_test.go]: Casting both to `[]byte` would be a simpler option

Comment (shiqizng) [reply api/handlers_e2e_test.go]: it seems that I'm only allowed to cast sprfKey to [64]byte because merklekeystore.Verifier is a type alias for [64]byte.

Comment (jasonpaulos) [reply api/handlers_e2e_test.go]: Oh good point. Will `sprfkey[:]` to convert to a slice?

Comment (shiqizng) [reply api/handlers_e2e_test.go]: updated.

**Review by tolikzinovyev — COMMENTED** [review]
LGTM. We will need to wait for state proof changes to arrive in go-algorand before merging.

**Review by winder — APPROVED** [review]
LGTM. But please don't merge it in yet, we're still finalizing a release.

File: api/handlers_test.go
Comment (algoidan) [reply api/handlers_test.go:640]: Adding this thing makes the CI fail.
do you have a different suggestion for that?

Comment (winder) [reply api/handlers_test.go:640]: Why is this function being called? It looks like a helper for adding a new test.

Comment (algoidan) [reply api/handlers_test.go:640]: yes
the idea is to keep the code that generates the test_resource.
I couldn't find more functions like this. Do you think it is worth keeping? (not as a Test function?)

Comment (winder) [reply api/handlers_test.go:640]: Yes, it is worth keeping.

`addNewTest` above is set to false, so this function is never called. It should never be called in a finished test which is why I suggested putting `assert.Fail` in here.

**Review by tolikzinovyev — APPROVED** [review]

**Pull request merged**
